### PR TITLE
improve sun/shadow example to use effect rendering pipeline

### PR DESCRIPTION
**Pull request by jianhuang01**
For #2993

#### Change List
- new sun/shadow experimental example
- create shadow-effect class
- refactor example to use effect rendering pipeline
- remove dummy shadow map
- delete pass and texture

### CODE REVIEW

**Comment by xintongxia** [comment]
Could you add a screenshot or .gif?

**Comment by jianhuang01** [comment]
image

File: examples/experimental/sun/src/app.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+// Source data GeoJSON
+const DATA_URL =
+  'https://raw.githubusercontent.com/uber-common/deck.gl-data/master/examples/trips/buildings.json'; // eslint-disable-line
+
+const ambientLight = new AmbientLight({
+  color: [255, 255, 255],
+  intensity: 1.0
+});
+
+const dirLight = new DirectionalLight({
```
Comment (Pessimistress) [inline examples/experimental/sun/src/app.js]: Use `SunLight`?

File: examples/experimental/sun/src/app.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+const material = new PhongMaterial({
+  ambient: 0.1,
+  diffuse: 0.6,
+  shininess: 32,
+  specularColor: [60, 64, 70]
+});
+
+const shadowColor = [2, 0, 5, 200];
+const shadowEffect = new ShadowEffect({shadowColor});
```
Comment (Pessimistress) [inline examples/experimental/sun/src/app.js]: Define inline: `new ShadowEffect({shadowColor: [2, 0, 5, 200]});`

File: examples/experimental/sun/src/shadow-effect/shadow-pass.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+      }
+    });
+    this.fbo = new Framebuffer(gl, {
+      id: 'shadowmap',
+      width: 1,
+      height: 1,
+      attachments: {
+        [gl.COLOR_ATTACHMENT0]: this.shadowMap,
+        // Depth attachment has to be specified for depth test to work
+        [gl.DEPTH_ATTACHMENT]: new Renderbuffer(gl, {
```
Comment (Pessimistress) [inline examples/experimental/sun/src/shadow-effect/shadow-pass.js]: This Renderbuffer needs to be deleted too

File: examples/experimental/sun/src/app.js (hunk 1 above)
Comment (jianhuang01) [reply examples/experimental/sun/src/app.js]: After integrating with SunLight, found the light direction is not right, need do more investigation on the sun location calculation code.

**Pull request merged**